{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1553, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-14-63527ed8e4e8>\", line 24, in getvalue\n",
      "    getdata = tx1.get()\n",
      "AttributeError: 'NoneType' object has no attribute 'get'\n"
     ]
    }
   ],
   "source": [
    "\n",
    "from tkinter import *\n",
    "\n",
    "#***********************************\n",
    "#Creates an instance of the class tkinter.Tk.\n",
    "#This creates what is called the \"root\" window. By conventon,\n",
    "#the root window in Tkinter is usually called \"root\",\n",
    "#but you are free to call it by any other name.\n",
    "\n",
    "root = Tk()\n",
    "root.geometry(\"500x500\")\n",
    "root.title('how to get text from textbox')\n",
    "\n",
    "\n",
    "mystring = StringVar()\n",
    "\n",
    "####define the function that the signup button will do\n",
    "#def getvalue():\n",
    "##    print(mystring.get())\n",
    "#*************************************\n",
    "\n",
    "def getvalue():\n",
    "   \n",
    "    getdata = tx1.get()\n",
    "    printdata = str(getdata)\n",
    "    tx2=Text(root).grid(row=0, column=1)\n",
    "    tx2.insert(END,printdata)\n",
    "    \n",
    "    \n",
    "l1=Label(root, text=\"Enter the input\").grid(row=0) #label\n",
    "tx1=Entry(root, textvariable = mystring).grid(row=0, column=1) #entry textbox\n",
    "\n",
    "\n",
    "bt1=Button(root,text=\"predict\", command=getvalue).grid(row=10,column=1)\n",
    "\n",
    "#WSignUp = Button(root, text=\"print text\", command=getvalue).grid(row=3, column=0, sticky=W) #button\n",
    "\n",
    "root.mainloop()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "from tkinter import *\n",
    "import tkinter as tk\n",
    "#from tkinter.tk \n",
    "#import *\n",
    "\n",
    "root = Tk()\n",
    "root.geometry(\"500x500\")\n",
    "\n",
    "label = Label(root, text=\"Hello World\")\n",
    "label.pack()\n",
    "mytext = tk.Text(root)\n",
    "mytext.insert('1.0', \"I love my text widget!\")\n",
    "mytext.pack()\n",
    "tvar = tk.StringVar()\n",
    "def swaptext():\n",
    "    tvar.set('There' if tvar.get() == 'Hi' else 'Hi')\n",
    "\n",
    "my_button = tk.Button(root, textvariable=tvar, command=swaptext)\n",
    "my_button.pack()\n",
    "\n",
    "root.mainloop()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "from tkinter import Frame, Label, Entry\n",
    "\n",
    "form = Frame()\n",
    "label = Label(form, text='Name')\n",
    "name_input = Entry(form)\n",
    "label.grid(row=0, column=0)\n",
    "name_input.grid(row=1, column=0)\n",
    "\n",
    "form.pack()\n",
    "form.mainloop()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
